pontoon/base/models.py

```diff
@@ -2330,6 +2331,79 @@ def cleaned_key(self):
     def __str__(self):
         return self.string
 
+    def get_stats(self, locale):
+        """
+        Get stats for a single (entity, locale) pair.
+
+        :arg Locale locale: filter translations for this locale.
+        :return: a dictionary with stats for an Entity, all keys are suffixed with `_diff` to
+            make them easier to pass into adjust_all_stats.
+        """
+        translations = list(
+            self.translation_set
+                .filter(locale=locale)
+                .prefetch_related(
+                    'errors',
+                    'warnings',
+                )
+        )
+
+        approved_strings_count = len([
+            t for t in translations
+            if t.approved and not(t.errors.exists() or t.warnings.exists())
+        ])
+        fuzzy_strings_count = len([
+            t for t in translations
+            if t.fuzzy and not(t.errors.exists() or t.warnings.exists())
+        ])
+        errors_count = len([
+            t for t in translations
+            if (t.approved or t.fuzzy) and t.errors.exists()
+        ])
+        warnings_count = len([
+            t for t in translations
+            if (t.approved or t.fuzzy) and t.warnings.exists()
+        ])
+        unreviewed_count = len([
+            t for t in translations
+            if not (t.approved or t.fuzzy or t.rejected)
+        ])
+
+        if self.string_plural:
+            approved = int(approved_strings_count == locale.nplurals)
+            fuzzy = int(fuzzy_strings_count == locale.nplurals)
+            if not (approved or fuzzy):
+                if errors_count:
+                    errors_count = 1
+                elif warnings_count:
+                    warnings_count = 1
+        else:
+            approved = int(approved_strings_count == 1)
+            fuzzy = int(fuzzy_strings_count == 1)
```

Comment:
It shouldn't be possible to have multiple approved/fuzzy translations for one `EntityLocale`, but to stay on a safe side, I'd change `== 1` to `> 0`.